nexus_constructor/model/load_from_json.py

```diff
@@ -53,9 +78,217 @@ def _retrieve_children_list(json_dict: dict) -> list:
         return []
 
 
+def _create_transformation_dataset(
+    angle_or_magnitude: float, dtype: str, name: str
+) -> Dataset:
+    """
+    Creates the transformation dataset using the angle/magnitude, name, and dtype of the transformation.
+    :param angle_or_magnitude: The angle or magnitude.
+    :param dtype: The data type.
+    :param name: The transformation name.
+    :return: A dataset containing the above information.
+    """
+    return Dataset(
+        name, dataset=DatasetMetadata(size=[1], type=dtype), values=angle_or_magnitude,
+    )
+
+
+class TransformationReader:
+    def __init__(self, parent_component: Component, entry: list):
+        self.parent_component = parent_component
+        self.entry = entry
+        self.warnings = []
+
+    def add_transformations_to_component(self):
+        """
+        Attempts to construct Transformation objects using information from the JSON dictionary and then add them to the
+        parent component.
+        """
+        for item in self.entry:
+            if _contains_transformations(item):
+                try:
+                    self._create_transformations(item["children"])
+                except KeyError:
+                    continue
+
+    def _get_transformation_property(
+        self,
+        property_name: str,
+        json_transformation: dict,
+        transform_name: str = None,
+        failure_value: Any = None,
+        parse_result_func: Callable = None,
+    ) -> Any:
+        """
+        Tries to find a certain property of a transformation from dictionary.
+        :param property_name: The name of the property fields.
+        :param json_transformation: The dictionary to look for the property in.
+        :param transform_name: The name of the transformation (if known).
+        :param failure_value: The value to return if the property cannot be found.
+        :param parse_result_func: A function that should be called on the dictionary value if it is found.
+        :return: Returns the property or converted property if this exists in the dictionary, if the property is not
+        found in the dictionary then the failure_value is returned.
+        """
+        try:
+            if not parse_result_func:
+                return json_transformation[property_name]
+            else:
+                return parse_result_func(json_transformation[property_name])
+        except KeyError:
+            if transform_name:
+                msg = (
+                    f"Cannot find {property_name} for transformation in component"
+                    f" {transform_name}"
+                )
+                f" {self.parent_component.name}."
+            else:
+                msg = f"Cannot find {property_name} for transformation in component"
+                f" {self.parent_component.name}."
+            self.warnings.append(msg)
+            return failure_value
+
+    def _find_property_in_list(
```

Comment:
If you pass the warning message string into the method then couldn't this method also be used to find "NX_class" instead of having a separate implementation for that (`_read_nx_class`)?

Reply:
There is some inconsistency in terminology, does "property" always mean "attribute"?
If yes then you could rename the method, if no then `attributes_list` could be renamed.